### PR TITLE
feat: Add decode with offsets

### PR DESCRIPTION
**Pull request by tomeras91**
Add a decode function to JurassicTokenizer that can return the offsets of the tokens in the string

### CODE REVIEW

File: ai21_tokenizer/jurassic_tokenizer.py [hunk 1]
```diff
@@ -174,6 +174,13 @@ def decode(self, token_ids: List[int], **kwargs) -> str:
         """
         Transforms token ids into text
         """
+        res_text, offsets = self.decode_with_offsets(token_ids)
```
Comment (asafgardin) [inline ai21_tokenizer/jurassic_tokenizer.py:177]: if `offsets` is not used in the response, you can do this instead `decoded_text, _ = ...`

File: ai21_tokenizer/jurassic_tokenizer.py [hunk 2]
```diff
@@ -174,6 +174,13 @@ def decode(self, token_ids: List[int], **kwargs) -> str:
...
+        return res_text
+
+    def decode_with_offsets(self, token_ids: List[int]) -> Tuple[str, List[Tuple[int, int]]]:
```
Comment (asafgardin) [inline ai21_tokenizer/jurassic_tokenizer.py:180]: Add a test please

File: tests/test_jurassic_tokenizer.py [hunk 3]
```diff
@@ -26,6 +26,21 @@ def test_tokenizer_encode_set(tokenizer: JurassicTokenizer, resources_path: Path
             ), f"Not equal at doc {i}"
 
 
+def test_tokenizer_decode_set(tokenizer: JurassicTokenizer, resources_path: Path):
```
Comment (asafgardin) [inline tests/test_jurassic_tokenizer.py:29]: `decode_with_offset` should be tested separately from `decode`
You don't have to test `decode` as you didn't add it

Comment (asafgardin) [reply tests/test_jurassic_tokenizer.py:29]: So simply rename this test and remove the call to `decode`

Comment (tomeras91) [reply tests/test_jurassic_tokenizer.py]: Done

**Review by asafgardin — APPROVED** [review]

**Pull request merged**